docs/api.md

```diff
@@ -0,0 +1,405 @@
+## Sematic decorator
```

Comment:
This doc doesn't look as good as generated API docs, but I guess it has to do.

Reply:
Exactly. It will do for now until we figure out a way to automate it.